examples/complex.c

```diff
@@ -80,6 +81,52 @@ static int certfail_handler(const xmpp_tlscert_t *cert,
     return read_char[0] == 'y' || read_char[0] == 'Y';
 }
 
+static int
+password_callback(char *pw, size_t pw_max, const char *fname, void *userdata)
```

Comment:
Quick question before reading the function: is it expected to implement this number of steps by users every time? From the comment I understand, that this code caches password, but as far as I understand, for a single connection this callback will always be called for the same key file, right? If so, can we move caching to libstrophe? I know, this is not secure, but we already cache account's password. Besides, we can clear the cached password after connection establishment.